src/main/java/de/tum/in/www1/artemis/service/iris/IrisGPT3_5Service.java

```diff
@@ -0,0 +1,202 @@
+     * explanation, see <a href="https://platform.openai.com/docs/api-reference/completions/create">...</a>
+     */
+    @Value("${artemis.iris.models.gpt3_5.temperature}")
+    private float temperature;
+
+    /**
+     * The maximum number of tokens that the model should generate in its responses. This is a safety measure to prevent
+     * the model from returning too much data.
+     */
+    @Value("${artemis.iris.models.gpt3_5.max-generated-tokens}")
+    private int maxGeneratedTokens;
+
+    /**
+     * Up to four sequences of characters that the model should stop generating text at. This is a safety measure to
+     * prevent the model from generating too much text.
+     */
+    @Value("${artemis.iris.models.gpt3_5.stop-sequences}")
+    private String[] stopSequences;
+
+    public IrisGPT3_5Service(RestTemplate restTemplate) {
+        this.restTemplate = restTemplate;
+    }
+
+    /**
+     * Prompts GPT-3.5 to generate a response to the conversation in the provided {@link IrisSession}.
+     *
+     * @param session The session to generate a response for.
+     * @return A {@link CompletableFuture} that will be completed with the response.
+     */
+    @Override
+    @Async
+    public CompletableFuture<String> getResponse(IrisSession session) {
+        try {
+            Map<String, Object> requestParams = createRequestParams(session.getMessages());
+            ObjectMapper jsonMapper = new ObjectMapper();
+            String requestBody = jsonMapper.writeValueAsString(requestParams);
+            log.debug("Request body: {}", requestBody);
+
+            HttpEntity<String> request = new HttpEntity<>(requestBody, createHeaders());
+            String responseBody = restTemplate.postForObject(apiURL.toString(), request, String.class);
+            log.debug("Response body: {}", responseBody);
+
+            Map<?, ?> response = jsonMapper.readValue(responseBody, Map.class);
+
+            // TODO: Handle errors if response is in error format
+            List<?> choices = (List<?>) response.get("choices");
+            Map<?, ?> choice = (Map<?, ?>) choices.get(0);
+            Map<?, ?> message = (Map<?, ?>) choice.get("message");
+            String content = (String) message.get("content");
+
+            return CompletableFuture.completedFuture(content);
+        }
+        catch (JsonProcessingException e) {
+            log.error(e.getMessage(), e);
+            return CompletableFuture.failedFuture(e);
+        }
+    }
+
+    /**
+     * Gets a {@link Map} of request parameters to be sent to GPT-3.5. These parameters will be converted into a
+     * JSON object and sent in the request body.
+     *
+     * @param conversationHistory The messages that have been sent in the conversation so far
+     * @return The API request parameters
+     */
+    private Map<String, Object> createRequestParams(List<IrisMessage> conversationHistory) {
+        Map<String, Object> parameters = new HashMap<>();
```

Comment:
Creating a DTO for this would probably also make sense for readability.
But you could also leave it like this for now.

Reply:
For now I think it is a small enough set of parameters that readability is not affected.